core/storage/suggestion/gae_models.py

```diff
@@ -1990,3 +2000,1275 @@ def export_data(
+
+        Returns:
+            3-tuple(sorted_results, next_offset, more). where:
+                sorted_results:
+                    list(TranslationSubmitterTotalContributionStatsModel).
+                    The list of models with supplied filters and sort_by.
+                next_offset: int. Number of results to skip in next batch.
+                more: bool. If True, there are (probably) more results after
+                    this batch. If False, there are no further results
+                    after this batch.
+        """
+
+        sort_options_dict = {
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingLastActivity']:
+                -cls.last_contribution_date,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingLastActivity']:
+                cls.last_contribution_date,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingPerformance']:
+                cls.recent_performance,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingAccuracy']:
+                -cls.overall_accuracy,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingAccuracy']:
+                cls.overall_accuracy,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['DecreasingSubmissions']:
+                -cls.submitted_translations_count,
+            constants.CD_ADMIN_STATS_SORT_OPTIONS['IncreasingSubmissions']:
+                cls.submitted_translations_count
+        }
+
+        sort = -cls.recent_performance
+        if sort_by is not None:
+            sort = sort_options_dict[sort_by]
+
+        # The first sort property must be the same as the property to which
+        # an inequality filter is applied. Thus, the inequality filter on
+        # last_activity can not be used here. Learn more about this here
+        # https://cloud.google.com/appengine/docs/legacy/standard/go111/datastore/query-restrictions#properties_used_in_inequality_filters_must_be_sorted_first.
+        if topic_ids is not None:
+            sort_query = cls.query(
+                datastore_services.all_of(
+                    cls.language_code == language_code,
+                    cls.topic_ids_with_translation_submissions.IN(topic_ids)
+                )).order(sort)
+        else:
+            sort_query = cls.query(
+                datastore_services.all_of(
+                    cls.language_code == language_code
+                )).order(sort)
+
+        sorted_results: List[
+            TranslationSubmitterTotalContributionStatsModel] = []
+        today = datetime.date.today()
+
+        if last_activity is not None:
+            last_date = today - datetime.timedelta(days=last_activity)
+            next_offset = offset
+            while len(sorted_results) < page_size:
+                result_models: Sequence[
+                    TranslationSubmitterTotalContributionStatsModel] = (
+                    sort_query.fetch(
+                        NUM_MODELS_PER_FETCH, offset=next_offset))
```

Comment:
I think you can make NUM_MODELS_PER_FETCH larger, say 500. Otherwise you'll potentially end up fetching quite a lot of pages in the case where you'll need to retrieve all the results. It's better to have fewer queries.

Reply:
Yeah sg. Changed it to 500.